### ISSUE TITLE
Possible wrong RandLA-Net implementation

### ISSUE THREAD

**Issue by saedrna** [torch-points3d/torch-points3d#782]
Hi,

Thanks for this awesome framework for deep learning of point clouds. I am now looking into the code of RandlaNet.
I found the attentive pooling might be wrong.

https://github.com/torch-points3d/torch-points3d/blob/66e8bf22b2d98adca804c753ac3f0013ff4ec731/torch_points3d/modules/RandLANet/modules.py#L28-L54

In message, features (`x` and `pos`) has dim of `(E,D)` (E for edge and D for channel) and the attention mapping `fij_hat` also has dim of `(E,D')`.
The `softmax` is applied to the feature dimension, it is like the channel attention pooling, rather than the positional attention pooling within the K nearest neighbors.

Because the inputs of the `message` function do not have information about the KNN at all. I am wondering how to implement such spatial pooling across the KNN points.

Or maybe I misunderstood the implementation?

Thanks,
Han

**Comment by CharlesGaydon** [comment]
@saedrna I have seen in answers from mainteners in several issues that the RandLa-Net impplementation would need a full rewrite and is not working at the moment (see https://github.com/torch-points3d/torch-points3d/issues/236#issuecomment-840418385)

I took some time to offer my own implementation of RandLa-Net direcly as an example in torch_geometric. Feel free to have a look on the currently opened [pull **request](https://github.com/pyg-team/pytorch_geometric/pull/5117).** Right now there may still be issues to be found with my implementation, which I hope we will manage to correct :)
